lib/stats.dx

```diff
@@ -16,11 +32,25 @@ instance {f} [Sub f] Fractional (LogSpace f)
 instance Arbitrary (LogSpace Float)
   arb = \k. Exp (randn k)
 
+def is_finite {f} [Fractional f, Sub f, Mul f, Ord f] (x:f) : Bool =
+  -- Todo: Make polymorphic versions of these and put them in the prelude.
+  infinity = (divide one zero)
+  neg_infinity = sub zero infinity
+  not (x == infinity || x == neg_infinity)
```

Comment:
Conventionally, is_finite nan should return False.  NaN can be checked for because it's the only floating-point number x satisfying `not (x == x)`.  Could you add that condition, and throw in a unit test for is_finite to make sure it stays correct in the future?

Reply:
Yes, will do.

Reply:
I was using this function in a way where I didn't want to include `nan`.  So I changed from `in_finite` to `is_infinite`, which doesn't include `nan`.  I added a comment and some tests to make the intended behavior clear.